File: InterviewBit/MergeIntervals.py
class Solution:
    def insert(self, a, b):
        p= b.start
        q = b.end
        res = []
        
        for k in range(len(a)):
            f = 0
            i = a[k].start
            j = a[k].end
            if p<=i and q>=j:
                f = 1
            if p>=i and p<=j:
                p = i
                f = 1
            if q>=i and q<=j:
                q = j
                f = 1
            if f == 1:
                continue
            res.append([i,j])
        res+=[[p,q]]
        res.sort()
        ans = []
        for i,j in res:
            ans.append(Interval(i,j))
        return ans

File: Leetcode/MergeIntervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& A) {
        sort(A.begin(),A.end());
        vector<vector<int>> res;
        for(int i = 0;i<A.size();i++){
            int n = res.size();
            if(res.empty() || res[n-1][1]<A[i][0])
                res.push_back({A[i][0],A[i][1]});
            else
                res[n-1][1] = max(res[n-1][1],A[i][1]);
            
        }
        return res;
    }
};